extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -126,6 +126,10 @@
     private static final Handler<HttpServerRequest> ACTUAL_ROOT = new Handler<HttpServerRequest>() {
         @Override
         public void handle(HttpServerRequest httpServerRequest) {
+            if (httpServerRequest.absoluteURI() == null) {
```

Comment:
I don't know about that. It will log something with a full stacktrace and I don't think we want that.

Given it's `new URI(httpServerRequest.uri())` that fails, why not do it ourselves and catch the exception? That would avoid the logging that I'm really not sure we want.

Reply:
Really we should just change vert.x to remove the logging. If we manually parse the URI ourselves then it will get parsed twice in most situations, which could affect performance.